src/plugins/query_enhancements/server/connections/clients/base_connection_client.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { TransportRequestParams } from '@opensearch-project/opensearch/lib/Transport';
+import { OpenSearchClient } from 'src/core/server';
+
+export interface ResourcesQuery {
+  dataSourceName: string; // SQL/DQS data source name (my_prometheus)
+  resourceType: string; // prometheus
+  // dataConnectionId: string; // or the API can take the data-connection saved object ID instead
+  query: Record<string, string>; // query params for raw prometheus API
+}
+
+export interface ClientRequest {
+  query: ResourcesQuery;
+  path: string;
+  params: Partial<TransportRequestParams>;
+}
+
+type ResourceData = unknown;
+
+export interface GetResourcesResponse {
+  nextToken?: string;
+  status: string;
+  data: ResourceData[];
+  type: string;
+}
+
+export abstract class BaseConnectionClient {
+  protected abstract client: OpenSearchClient;
```

Comment:
this might not be true for all concrete connection clients, for `SqlConnectionClient` we'll use `OpenSearchClient` to make the request, but it can be other underlying client for other concrete connection clients

Reply:
Makes sense. I can have this client interface be a generic that the specific connection clients can provide.